test/functional/helpers/createFixture/clientScripts.js

```diff
@@ -0,0 +1,206 @@
+// is "int". Otherwise, we're testing against the prod alloy that's on the CDN
+// and in those cases alloy code may not event exist in /dist. This is why
+// we only try to load the file from the file system if alloyEnv is int.
+if (alloyEnv === "int") {
+  localAlloyCode = fs.readFileSync(localAlloyLibraryPath, "utf8");
+}
+
+const addRemoteUrlClientScript = ({ clientScripts, url, async = false }) => {
+  // TestCafe client scripts don't "natively" support loading a script
+  // from a remote URL, so we have to make our local script add another script
+  // element that loads Visitor from the remote server.
+  clientScripts.push({
+    content: `document.write('<script src="${url}"${
+      async ? " async" : ""
+    }></script>')`
+  });
+};
+
+/**
+ * Produces an array of scripts that TestCafe should inject into the <head>
+ * when testing Alloy int.
+ */
+const getFixtureClientScriptsForInt = options => {
+  const clientScripts = [];
+
+  // We load a promise polyfill because promises aren't supported in IE
+  // and that's what customers supporting IE will need to do as part
+  // of installing Alloy.
+  // We could load the Promise polyfill from a CDN, but it would slow down
+  // our tests a bit since we would have to go over the wire to load it.
+  clientScripts.push({
+    path: localPromisePolyfillPath
+  });
+
+  // Useful for testing Alloy + Visitor interaction.
+  if (options.includeVisitorLibrary) {
+    addRemoteUrlClientScript({
+      clientScripts,
+      url: remoteVisitorLibraryUrl
+    });
+  }
+
+  clientScripts.push({
+    path: alloyPageSnippetPath
+  });
+
+  // Typically the Alloy library should be loaded in head. For some tests,
+  // like testing command queuing, we need greater control and will
+  // load the Alloy library later during the test using injectAlloyDuringTest.
+  if (options.includeAlloyLibrary) {
+    // When providing client scripts to TestCafe during the fixture
+    // configuration process, TestCafe doesn't currently support loading scripts
+    // asynchronously (https://github.com/DevExpress/testcafe/issues/5612), but
+    // by default our customers are loading Alloy asynchronously and we would
+    // like to simulate that. To do so, we'll wrap our Alloy code in a
+    // setTimeout with a small arbitrary delay.
+    clientScripts.push({
+      content: `setTimeout(() => {\n${localAlloyCode}\n}, 10);`
+    });
+  }
+
+  return clientScripts;
+};
+
+/**
+ * Produces an array of scripts that TestCafe should inject into the <head>
+ * when testing Alloy prod.
+ */
+const getFixtureClientScriptsForProd = options => {
+  const clientScripts = [];
+  addRemoteUrlClientScript({
+    clientScripts,
+    url: remotePromisePolyfillPath
+  });
+
+  // Useful for testing Alloy + Visitor interaction.
+  if (options.includeVisitorLibrary) {
+    addRemoteUrlClientScript({
+      clientScripts,
+      url: remoteVisitorLibraryUrl
+    });
+  }
+
+  clientScripts.push({
+    path: alloyPageSnippetPath
+  });
```

Comment:
Note that we're always using the alloy page snippet that's in the file committed to the repo. There's not really a "production" page snippet.

Reply:
Why not use the page snippet that is built into dist/ ?

Reply:
Yeah, that's a great question. One thing to note is that the snippet isn't ever build to `dist`. We do have a `build:basecode` npm script that takes `/src/baseCode/index.js`, runs it through a minifier, and outputs the result to the console. The idea is we'd copy it out of the console to update the documentation, for example.

But....that doesn't mean we couldn't change our script to dump a minified version into `dist` for at least functional test purposes. Or, more simply, we could just read the unminified code from `/src/baseCode/index.js` and use it. Part of the reason that wasn't done, I presume, was that the snippet we use for functional tests isn't the same as the snippet that's in `/src/baseCode/index.js` or that's produced by `build:basecode`. The difference is that the snippet used for functional tests configures two instances: `alloy` _and_ `instance2`.

That said, I'm thinking it might be best if, for functional testing, we read the base code from `/src/baseCode/index.js` and then just did a simple string replace to modify `["alloy"]` to be `["alloy", "instance2"]`. Thoughts?

Reply:
Unless someone is against the approach I proposed, I'm going to make it happen, but I'm going to do it in a separate PR just so we can get this PR merged to master so @liljenback can start building on top of it.